php/Factory.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace PHPCD;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\DependencyInjection\Loader\YamlFileLoader;
+use Lvht\MsgpackRpc\JsonMessenger;
+use Lvht\MsgpackRpc\Io;
+use Lvht\MsgpackRpc\MsgpackMessenger;
+
+/**
+ * Simple factory to separate details of object creation
+ */
+class Factory
+{
+    /**
+     * @return ContainerBuilder
+     */
+    public function createDIContainer($configFileName, $configDir, $additionalParameters = [])
+    {
+        $container = new ContainerBuilder;
+
+        $loader = new YamlFileLoader($container, new FileLocator($configDir));
```

Comment:
> In my opinion, there is no need to introduce the DI. I make the #47 to drop them. FYI.

You probably thought about DI container, not DI itself. So called DI container may be used (usually by the wrong way) without dependency injection. On the other side, we can inject dependencies manually without using any container (as you made in #47 except instantiating some objects from inside).